src/server/frames.ts

```diff
@@ -556,8 +557,9 @@ export class Frame extends SdkObject {
   }
 
   async goto(metadata: CallMetadata, url: string, options: types.GotoOptions = {}): Promise<network.Response | null> {
+    const constructedNavigationUrl = constructNavigationUrl(this._page._browserContext._options.baseUrl, url);
```

Comment:
I am thinking whether any other urls we have should follow this pattern? `waitForURL`, `waitForRequest`, `route`, something else?

Reply:
On navigation is in my mind the most crucial part. For route I don't want to have more magic, we already have glob/regexp and predicates there IMO.